CactusCare.BLL/Services/ReviewService.cs

```diff
@@ -36,18 +37,28 @@ public async Task InsertAsync(ReviewDTO reviewDto)
         {
             await _unitOfWork.ReviewRepository.InsertAsync(_mapper.Map<ReviewDTO, Review>(reviewDto));
             await _unitOfWork.SaveAsync();
+            await UpdateDoctorRatingAsync(reviewDto.DoctorId);
         }
 
         public async Task UpdateAsync(ReviewDTO reviewDto)
         {
             await _unitOfWork.ReviewRepository.UpdateAsync(_mapper.Map<ReviewDTO, Review>(reviewDto));
             await _unitOfWork.SaveAsync();
+            await UpdateDoctorRatingAsync(reviewDto.DoctorId);
```

Comment:
you called SaveAsync two times here, it's bad for performance and doesn't follow ACID principles